rules/type-error.js

```diff
@@ -0,0 +1,106 @@
+'use strict';
+
+const tcIdentifiers = new Set([
+	'isArguments',
+	'isArray',
+	'isArrayBuffer',
+	'isArrayLike',
+	'isArrayLikeObject',
+	'isBoolean',
+	'isBuffer',
+	'isDate',
+	'isElement',
+	'isEmptyObject',
+	'isError',
+	'isFinite',
+	'isFrozen',
+	'isFunction',
+	'isInteger',
+	'isLength',
+	'isMap',
+	'isNaN',
+	'isNative',
+	'isNil',
+	'isNull',
+	'isNumber',
+	'isObject',
+	'isObjectLike',
+	'isPlainObject',
+	'isPrototypeOf',
+	'isRegExp',
+	'isSafeInteger',
+	'isSealed',
+	'isSet',
+	'isString',
+	'isSymbol',
+	'isTypedArray',
+	'isUndefined',
+	'isView',
+	'isWeakMap',
+	'isWeakSet',
+	'isWindow',
+	'isXMLDoc'
+]);
+
+const isTypecheckingProvider = node => node.type === 'Identifier' && false;
+const isTypecheckingOperator = operator => operator === 'typeof' || operator === 'instanceof';
+const isTypecheckingIdentifier = node => node.type === 'Identifier' && tcIdentifiers.has(node.name);
+
+const throwsErrorObject = node =>
+	node.argument.type === 'NewExpression' &&
+	node.argument.callee.type === 'Identifier' &&
+	node.argument.callee.name === 'Error';
+
+const isTypecheckingMemberExpression = node => {
```

Comment:
I think this function considers `isNaN.foo()` and `isNaN` (without being called) as type checking expressions. Add those as valid tests